flask_app/models.py

```diff
@@ -95,11 +108,26 @@ class TestMetadata(db.Model):
     metadata_item = db.Column(JSONB)
 
 
-class Error(db.Model):
+class SessionMetadata(db.Model):
 
     id = db.Column(db.Integer, primary_key=True)
+    session_id = db.Column(db.Integer, db.ForeignKey('session.id', ondelete='CASCADE'), index=True)
+    metadata_item = db.Column(JSONB)
+
+
+class TestError(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
     test_id = db.Column(db.Integer, db.ForeignKey('test.id', ondelete='CASCADE'), index=True)
     traceback = db.Column(JSON)
     exception_type = db.Column(db.String(256), index=True)
     exception = db.Column(db.String(256), index=True)
+    timestamp = db.Column(db.Float, default=get_current_time)
+
+
+class SessionError(db.Model):
```

Comment:
That's what I was talking about in the email the other day, it requires
polymorphism (test_id, session_id - in that table)
Which is the "correct" approach but in SQLAlchemy means tons of code

On Tue, Apr 28, 2015 at 4:29 PM, Rotem Yaari <notifications@github.com>
wrote:

> In flask_app/models.py
> <https://github.com/slash-testing/backslash/pull/13#discussion_r29241738>:
>
> >      test_id = db.Column(db.Integer, db.ForeignKey('test.id', ondelete='CASCADE'), index=True)
> >      traceback = db.Column(JSON)
> >      exception_type = db.Column(db.String(256), index=True)
> >      exception = db.Column(db.String(256), index=True)
> > +    timestamp = db.Column(db.Float, default=get_current_time)
> > +
> > +
> > +class SessionError(db.Model):
>
> These two should be a single table (Error) and two many-to-many tables
> linking errors to sessions and tests respectively. Code duplication here
> will likely get out of sync...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/slash-testing/backslash/pull/13/files#r29241738>.
>